bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiRpc.java

```diff
@@ -700,10 +700,14 @@ public void onMessage(String message) {
     }
 
     @Override
-    public void onClose(int statusCode, String reason) {
+    public void onClose(int statusCode, String description) {
         try {
-            logger.debug("{}: WebSocket connection closed, status = {}/{}", thingName, statusCode, getString(reason));
-            if (statusCode == StatusCode.ABNORMAL && !discovery && getProfile().alwaysOn) { // e.g. device rebooted
+            String reason = getString(description);
+            logger.debug("{}: WebSocket connection closed, status = {}/{}", thingName, statusCode, reason);
+            if ("Bye".equalsIgnoreCase(reason)) {
+                logger.debug("{}: Device went to sleep mode", thingName);
```

Comment:
Just a check, if the device is in sleep mode, it is still responsive? As in you don't need to set the Thing status?

Reply:
No, when it sleeps it needs an event to wake-up  (sensors are still active, but the device is not accessible via the APII), e.g. the temp sensor reports a change then the device wakes up and sends an event message to the binding.

Maybe I should change the message to "Connection terminated, e.g. device in sleep mode"?

Reply:
@lsiepel changed